subsys/caf/modules/caf_shell.c

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdlib.h>
+#include <zephyr/shell/shell.h>
+#include <caf/events/button_event.h>
+
+static const char set_button_cmd_help_str[] =
+	"Submit a button_event with user-defined key ID and pressed state\n"
+	"  Key ID           Decimal numeric ID of the button\n"
+	"  Pressed state    'y' or 'n'\n";
+
+static int button_event_handler(const struct shell *shell, size_t argc, char **argv)
+{
+	/* The string part of button_id param, for invalidation check */
+	char *str_part;
+
+	/* Convert button_id from string to decimal number */
+	long button_id = strtol(argv[1], &str_part, 10);
```

Comment:
What do you think about using `0` instead of `10` as base to allow user to input values in hex too?

Reply:
btw. Should we check if the `button_id` actually fits in `uint16_t` (lower than `UINT16_MAX`) and is not negative?